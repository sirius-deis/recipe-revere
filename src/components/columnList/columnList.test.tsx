import { render, screen } from "@testing-library/react";
import ColumnList from "./columnList";

describe("ColumnList component", () => {
  const mockData = [<div>item 1</div>, <div>item 2</div>, <div>item 3</div>, <div>item 4</div>];
  it("should match snapshot", () => {
    const { container } = render(<ColumnList items={mockData} />);
    expect(container).toMatchSnapshot();
  })
  it("should render items in a grid layout with default number of columns", () => {
    render(<ColumnList items={[]} />);
    const columnList = screen.getByRole("list");
    expect(columnList).toHaveStyle("grid-template-columns: repeat(3, 1fr)");
  });

  it("should render items in a grid layout", () => {
    render(<ColumnList items={mockData} numberOfColumns={2} />);
    const columnList = screen.getByRole("list");
    expect(columnList).toHaveStyle("grid-template-columns: repeat(2, 1fr)");
    const items = screen.getAllByRole("list-item");
    expect(items.length).toBe(mockData.length)
  })
  it("should render items in a grid layout when numberOfColumns exceeds 3", () => {
    render(<ColumnList items={[]} numberOfColumns={4} />);
    const columnList = screen.getByRole("list");
    expect(columnList).toHaveStyle("grid-template-columns: repeat(3, 1fr)");
  })
})