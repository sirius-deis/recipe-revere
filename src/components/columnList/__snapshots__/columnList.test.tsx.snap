// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnList component should match snapshot 1`] = `
<div>
  <div
    class="column-list"
    role="list"
    style="grid-template-columns: repeat(3, 1fr);"
  >
    <div
      class="list-item"
      role="list-item"
    >
      <div>
        item 1
      </div>
    </div>
    <div
      class="list-item"
      role="list-item"
    >
      <div>
        item 2
      </div>
    </div>
    <div
      class="list-item"
      role="list-item"
    >
      <div>
        item 3
      </div>
    </div>
    <div
      class="list-item"
      role="list-item"
    >
      <div>
        item 4
      </div>
    </div>
  </div>
</div>
`;
